src/state.rs

```diff
@@ -183,26 +183,28 @@ impl BarState {
     }
 
     pub(crate) fn draw(&mut self, mut force_draw: bool, now: Instant) -> io::Result<()> {
-        let width = self.draw_target.width();
-
         // `|= self.is_finished()` should not be needed here, but we used to always draw for
         // finished progress bars, so it's kept as to not cause compatibility issues in weird cases.
         force_draw |= self.state.is_finished();
-        let mut drawable = match self.draw_target.drawable(force_draw, now) {
-            Some(drawable) => drawable,
-            None => return Ok(()),
-        };
 
-        let mut draw_state = drawable.state();
+        // First check whether there is something to be drawn.
+        if self.draw_target.drawable(force_draw, now).is_none() {
```

Comment:
This double call to `drawable` is sort of ugly in my eyes.

However, the returned `drawable` prevents us from calling `self.draw_target.width()`, since the mutable borrow is still active.